storage/innobase/include/dict0dict.ic

```diff
@@ -1813,7 +1813,7 @@ dict_table_get_index_on_first_col(
         ulint                   col_index)	/*!< in: position of column
 						in table */
 {
-	ut_ad(col_index < table->n_cols);
+	ut_a(col_index < table->n_t_cols);
```

Comment:
I think it is better to keep it as debug-only assert in 5.7 (keep it ut_ad()). Introducing new assert in non-debug builds in 5.7 at this stage doesn't sound like a good idea from risk management point of view to me.

Reply:
And there is more high-level question about why are we backporting fix for something that affects only debug builds.

Reply:
I am not quite convinced about backporting some other patches either. I guess we need to have a discussion about criteria for such backports.

But patches themselves look OK to me.